via/static/scripts/video_player/components/FilterInput.tsx

```diff
@@ -0,0 +1,52 @@
+import { CancelIcon, IconButton, Input } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { Ref } from 'preact';
+
+export type FilterInputProps = {
+  elementRef: Ref<HTMLElement | undefined>;
+  setFilter: (filter: string) => void;
+  filter: string;
+};
+
+export default function FilterInput({
+  elementRef,
+  setFilter,
+  filter,
+}: FilterInputProps) {
+  return (
+    <div className="relative">
+      <Input
+        data-testid="filter-input"
+        aria-label="Transcript filter"
+        classes={classnames(
+          // Match height of search input in sidebar
+          'h-[32px]',
+          // Extra padding right to prevent text and "clear" button overlapping
+          'pr-8'
+        )}
+        elementRef={elementRef}
+        onKeyUp={e => {
+          // Allow user to easily remove focus from search input.
+          if (e.key === 'Escape') {
+            (e.target as HTMLElement).blur();
+          }
+        }}
+        onInput={e => setFilter((e.target as HTMLInputElement).value)}
+        placeholder="Search..."
+        value={filter}
+      />
+      {filter && (
+        <IconButton
+          icon={CancelIcon}
+          data-testid="clear-filter-button"
+          title="Clear filter"
+          onClick={() => setFilter('')}
+          classes={classnames(
+            'focus-visible-ring',
```

Comment:
Shouldn't need this class (you get it for free).

Reply:
Done